kedro-telemetry/tests/test_plugin.py

```diff
@@ -258,3 +280,16 @@ def test_check_for_telemetry_consent_file_invalid_yaml(self, mocker, fake_metada
 
         assert _check_for_telemetry_consent(fake_metadata.project_path)
         mock_create_file.assert_called_once_with(telemetry_file_path)
+
+    def test_confirm_consent_yaml_dump_error(self, mocker, fake_metadata, caplog):
+        Path(fake_metadata.project_path, "conf").mkdir(parents=True)
+        telemetry_file_path = fake_metadata.project_path / ".telemetry"
+        mocker.patch("yaml.dump", side_effect=Exception)
+
+        assert not _confirm_consent(telemetry_file_path)
+
+        msg = (
+            "Failed to confirm consent. No data was sent to Heap. Exception: "
+            "pytest: reading from stdin while output is captured!  Consider using `-s`."
```

Comment:
It should be fine just checking this part "Failed to confirm consent. No data was sent to Heap." is included in the message.